snapcraft/meta/snap_yaml.py

```diff
@@ -558,3 +561,26 @@ def _populate_environment(
         del environment["PATH"]
 
     return environment if environment else None
+
+
+def _process_components(components: Dict[str, Any] | None) -> Dict[str, Any] | None:
```

Comment:
why do we even add a version if we are going to remove it?
is the type not a dict of str, Component?

Reply:
Thanks, I made the type more specific.

From my interpretation of the spec, a component's version belongs in the component's metadata but not in the snap's metadata.  I asked in the spec to see if my understanding is correct.